src/Microsoft.Extensions.Configuration.Abstractions/IConfigurationSource.cs

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.Extensions.Configuration
+{
+    /// <summary>
+    /// Represents a source of configuration key/values for an application.
```

Comment:
Isn't this exactly the description we had on `IConfigurationProvider`? Let's try to come up with a descriptions that are clear and more distinguishable. E.g. this type only represents a source of configuration in the builder.